com.ibm.streamsx.topology/opt/python/packages/streamsx/topology/tester.py

```diff
@@ -269,16 +269,17 @@ def _standalone_test(self, config):
         return sr['return_code'] == 0
 
     def _distributed_test(self, config, username, password):
-        sjr = stc.submit(stc.ContextTypes.DISTRIBUTED, self.topology, config, username=username, password=password)
-        self.submission_result = sjr
-        if sjr['return_code'] != 0:
-            print("DO AS LOGGER", "Failed to submit job to distributed instance.")
-            return False
         self.sc = config.get(ConfigParams.STREAMS_CONNECTION)
         if self.sc is None:
             # Disable SSL verification
             self.sc = StreamsConnection(username, password)
             self.sc.session.verify = False
+            config[ConfigParams.STREAMS_CONNECTION] = self.sc
+        sjr = stc.submit(stc.ContextTypes.DISTRIBUTED, self.topology, config, username=username, password=password)
```

Comment:
The username/password is needed to handle the case when user explicitly specified a username and password on `test()`.  In which case, the `StreamsConnection` would need to be setup with the same credential.